GetStream/stream-chat-android
# Improve asking for write external storage permission

### 🎯 Goal
Improve asking for write external storage permission.

### 🛠 Implementation details
We are using external storage (download directory) when downloading attachments using `ChatClient::downloadAttachment` extension function.
`WRITE_EXTERNAL_STORAGE` permission is not needed on Android > Android Q if we are accessing external storage directories like `DOWNLOAD`.
The PR includes 3 changes:
- Changed `WRITE_EXTERNAL_STORAGE` permission `android:maxSdkVersion` to 29
- Don't ask for permission if the version is > 29
- Ask for permission if the version is 29 and legacy external storage is being used.

### 🧪 Testing

You need to test 4 different scenarios. Each of them consists on:
1. Open the channel
2. Clik some image
3. Click the save button
4. Confirm the image is appropriately saved

Scenarios:
1. Run apps on API < 29
2. Run apps on API 29
3. Run apps on API 29 and apply the patch (it requests legacy storage)
4. Run apps on latest available API

_Provide a patch below if it is necessary for testing_

<details>

<summary>Provide the patch summary here</summary>

```
Index: stream-chat-android-ui-components/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/AndroidManifest.xml b/stream-chat-android-ui-components/src/main/AndroidManifest.xml
--- a/stream-chat-android-ui-components/src/main/AndroidManifest.xml	(revision ba402f507bbcab9205823705f67f4a772dac7506)
+++ b/stream-chat-android-ui-components/src/main/AndroidManifest.xml	(date 1664453914114)
@@ -21,7 +21,7 @@
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 
-    <application>
+    <application android:requestLegacyExternalStorage="true">
         <provider
             android:name="androidx.startup.InitializationProvider"
             android:authorities="${applicationId}.androidx-startup"
@@ -36,18 +36,18 @@
 
         <activity
             android:name=".gallery.AttachmentGalleryActivity"
-            android:theme="@style/StreamUi.AttachmentGallery"
             android:exported="false"
+            android:theme="@style/StreamUi.AttachmentGallery"
             />
         <activity
             android:name=".gallery.AttachmentMediaActivity"
-            android:theme="@style/StreamUi.MediaActivity"
             android:exported="false"
+            android:theme="@style/StreamUi.MediaActivity"
             />
         <activity
             android:name=".gallery.AttachmentActivity"
+            android:exported="false"
             android:theme="@style/StreamUi.AttachmentActivity"
-            android:exported="false"
             />
 
         <activity
Index: stream-chat-android-compose-sample/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/AndroidManifest.xml b/stream-chat-android-compose-sample/src/main/AndroidManifest.xml
--- a/stream-chat-android-compose-sample/src/main/AndroidManifest.xml	(revision ba402f507bbcab9205823705f67f4a772dac7506)
+++ b/stream-chat-android-compose-sample/src/main/AndroidManifest.xml	(date 1664453914110)
@@ -23,6 +23,7 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
+        android:requestLegacyExternalStorage="true"
         android:networkSecurityConfig="@xml/network_security_config"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"

```

</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
